src/overlay/Peer.cpp

```diff
@@ -852,6 +854,16 @@ Peer::recvError(StellarMessage const& msg)
 void
 Peer::noteHandshakeSuccessInPeerRecord()
 {
+    if (getIP().empty() || getRemoteListeningPort() == 0)
```

Comment:
noteHandshakeSuccessInPeerRecord() is called when we received auth so we are connected

getRemoteListeningPort() returns value that we got form peer in HELLO message which is set to config.PEER_PORT which have to be  > 0 (see Config.cpp:188).

So the only case it can be 0 here is some kind of logic error where we are trying to save PeerRecord to database before HELLO was received.